### DISCUSSION TITLE
Import LED Database into iped-hashes.db

### DISCUSSION THREAD

**Discussion by eduardonmarques** [sepinf-inc/IPED#1234] [General]
Hi, 
I started to import several databses to iped-hashes.db (succesfully done with ProjectVIc and all NSRL databases, resulting up to now in a 37,2GB size database)
Despite of this, I can´t import LED database from wkff folder using the command line below:
jre\bin\java -jar lib/iped-hashdb.jar -d C:\Kff_IPED\pedo\wkff -o C:\IPED\iped-hashes.db
When I do this way  the result is:
"C:\Kff_IPED\pedo\wkff\<filename>.txt skipped" for each .txt file inside the folder wkff
When I use the cmd line with a specific txt file 
jre\bin\java -jar lib/iped-hashdb.jar -d C:\Kff_IPED\pedo\wkff\laudo-0011-2016-setec-sp.txt -o C:\IPED\iped-hashes.db
the result is:
laudo-0011-2016-setec-sp.txt header must contain at least one hash column

What do I have to do to import LED database to hashes.db?

tks

**Comment by lfcnassif** [comment]
It is not supported directly anymore, it should be converted to a standard CSV format before, take a look at the manual.

**Reply by lfcnassif** [reply]
Maybe using some of the new parameters of hashdb tool (like csv separator, column header renaming, etc) it might be possible to import directly, take a look at the tool help.

**Comment by eduardonmarques** [comment]
thank you,

I read Manual and as far as I understood "LED Child Porn database" is ready to be imported to iped-hashes.db, instead of trying to import all hash files from wkff folder, as I was doing. Is it correct?

If so, where can I find LED Child Porn Database?

**Reply by lfcnassif** [reply]
I updated/fixed the manual right before my post to state it is not supported directly today, it must be converted to a standard CSV before importing. I'm on vacation now, otherwise I could take a look at and test importing tool advanced options to see if it is possible to import without converting to CSV before.

If you are law enforcement, you could ask for the ready to import LED CSV files at sepinf DOT ditec at pf DOT gov DOT br

**Comment by eduardonmarques** [comment]
Tks Nassif
I am also law enforcement. 
Now I found it at webdav...

**Comment by DHoelz** [comment]
@lfcnassif sorry for hijacking this post but I'm also having problems adding LED's hashes list to IPED version 4.

I created a csv file with all the hashes and added the four new columns (set, status, ledGroup and ledCategory). 

Importing the csv file went out ok, however when I try to process any case with LedCarveTask turned on I get the message "LED hashes must be loaded into IPED hashes database to enable LedCarveTask".

Tried to simplify the csv file with only the main columns (as shown below) but still got the same error. 
MD5,Edonkey,SHA1,SHA256,File Name,set,status,ledGroup,ledCategory

Inspecting the sqlite it seems it  was populated  properly.

Could you send me a one line sample, or an example,  to see what I'm doing wrong?

**Reply by wladimirleite** [reply]
Hi @DHoelz!

First of all, you can access our internal WebDAV, right? 
There is a CSV file there with all LED database hashes and a SQLite ready to use with all hashes from LED and other hash sets already imported.

Back to your question, here is the header and a sample content line of the CSV:
I guess that one or more of the required columns are missing in the CSV you created.
```
MD5,SHA1,EDONKEY,SHA256,status,set,md5_512,md5_64k,fileLength,fileExt,ledGroup,ledCategory
000001212D1A0312FF896AE308458412,5C187220CAC8D599FBEDF2D6196C36CFE2AE1E39,B531A964548EE4F4322257FBC1A5D25B,492C49FD7526DAFB9DCF327F887AAAE633CF572B1208B0BC5B029D960702308E,pedo,LED,E0695993EBD0BC0E210071A348C66484,422F10EE9CA4D17DF912FFDE879C62F4,250443,jpg,laudo-0069-2017-setec-sp,1
```

**Reply by DHoelz** [reply]
Hi @tc-wleite, thanks for the fast reply.

I'm from PCDF. So, no, I don't have access to your WEBDAV, unfortunately.

I'll try to import just this sample line you sent and see if it works.
